rasa/core/brokers/kafka.py

```diff
@@ -124,67 +124,73 @@ def publish(
+                "security.protocol": self.security_protocol.lower(),
+            }
         elif self.security_protocol == "SASL_PLAINTEXT":
-            authentication_params = dict(
-                sasl_plain_username=self.sasl_username,
-                sasl_plain_password=self.sasl_password,
-                sasl_mechanism=self.sasl_mechanism,
-                security_protocol=self.security_protocol,
-            )
+            authentication_params = {
+                "sasl.username": self.sasl_username,
+                "sasl.password": self.sasl_password,
+                "sasl.mechanism": self.sasl_mechanism,
+                "security.protocol": self.security_protocol.lower(),
+            }
         elif self.security_protocol == "SSL":
-            authentication_params = dict(
-                ssl_cafile=self.ssl_cafile,
-                ssl_certfile=self.ssl_certfile,
-                ssl_keyfile=self.ssl_keyfile,
-                ssl_check_hostname=False,
-                security_protocol=self.security_protocol,
-            )
+            authentication_params = {
+                "ssl.ca.location": self.ssl_cafile,
+                "ssl.certificate.location": self.ssl_certfile,
+                "ssl.key.location": self.ssl_keyfile,
+                "security.protocol": self.security_protocol.lower(),
+            }
         elif self.security_protocol == "SASL_SSL":
-            authentication_params = dict(
-                sasl_plain_username=self.sasl_username,
-                sasl_plain_password=self.sasl_password,
-                ssl_cafile=self.ssl_cafile,
-                ssl_certfile=self.ssl_certfile,
-                ssl_keyfile=self.ssl_keyfile,
-                ssl_check_hostname=self.ssl_check_hostname,
-                security_protocol=self.security_protocol,
-                sasl_mechanism=self.sasl_mechanism,
-            )
+            authentication_params = {
+                "sasl.username": self.sasl_username,
+                "sasl.password": self.sasl_password,
+                "ssl.ca.location": self.ssl_cafile,
+                "ssl.certificate.location": self.ssl_certfile,
+                "ssl.key.location": self.ssl_keyfile,
+                "ssl.endpoint.identification.algorithm": self.ssl_check_hostname,
+                "security.protocol": self.security_protocol.lower(),
+                "sasl.mechanism": self.sasl_mechanism,
+            }
         else:
             raise ValueError(
                 f"Cannot initialise `KafkaEventBroker`: "
                 f"Invalid `security_protocol` ('{self.security_protocol}')."
             )
 
         try:
-            return kafka.KafkaProducer(
-                client_id=self.client_id,
-                bootstrap_servers=self.url,
-                value_serializer=lambda v: json.dumps(v).encode(DEFAULT_ENCODING),
-                **authentication_params,
+            return confluent_kafka.SerializingProducer(
```

Comment:
Note that the `SerializingProducer` is marked as experimental in the [API docs](https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#confluent_kafka.SerializingProducer), but it was the only class accepting the `value.serializer` as config.

Reply:
i am fine with using it, but we can also replace it by just serializing the value before passing it to `produce`. fine either way.

Reply:
Replaced with `Producer` API: https://github.com/RasaHQ/rasa/pull/11765/commits/ad960d46c115ca8433779f2ab8e539d2449cac92